src/Gaufrette/Adapter/Ftp.php

```diff
@@ -366,7 +377,22 @@ private function isDir($directory)
             return true;
         }
 
-        if (!@ftp_chdir($this->getConnection(), $directory)) {
+        $chDirResult = false;
+        try {
+            $chDirResult = ftp_chdir($this->getConnection(), $directory);
```

Comment:
can you also move the change to the original directory here ?
```php
// change directory again to return in the base directory
ftp_chdir($this->getConnection(), $this->directory);
```
as it'd keep the `ftp_chdir` logic in this `try` block :

```php
try {
    $isDir = ftp_chdir($this->getConnection(), $directory);
    // change directory again to return in the base directory
    ftp_chdir($this->getConnection(), $this->directory);

    return $isDir;
}
```